File: 953-reverse-only-letters/reverse-only-letters.py
class Solution:
    def reverseOnlyLetters(self, s: str) -> str:
        stack = []
        res = []
        for i in range(len(s)):
            code = ord(s[i])
            if (65 <= code <= 90) or (97 <= code <= 122):
                stack.append(s[i])

        for i in range(len(s)):
            code = ord(s[i])
            if (65 <= code <= 90) or (97 <= code <= 122):
                res.append(stack.pop())
            else:
                res.append(s[i])

        return ''.join(res)

File: 953-reverse-only-letters/reverse-only-letters.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseOnlyLetters = function (s) {
    let stack = []
    let res = []
    for (let i = 0; i < s.length; i++) {
        let code = s.charCodeAt(i)
        if ((code >= 65 && code <= 90) || (code >= 97 && code <= 122)) {
            stack.push(s.charAt(i))
        }
    }
    for (let i = 0; i < s.length; i++) {
        let code = s.charCodeAt(i)
        if ((code >= 65 && code <= 90) || (code >= 97 && code <= 122)) {
            res.push(stack.pop())
        }
        else {
            res.push(s.charAt(i))
        }
    }
    return res.join("")
};